### ISSUE TITLE
SVC (kernel=linear) JS prediction logic

### ISSUE THREAD

**Issue by anmolshkl** [nok/sklearn-porter#22]
Hi @nok, the [libsvm ](https://github.com/cjlin1/libsvm/blob/master/svm.cpp#L2555) implementation seems to be using subtraction while the sklearn-porter's JavaScript [predict method](https://github.com/nok/sklearn-porter/blob/6da2f7cbaa885b044aa6fe29ee896be089965c54/sklearn_porter/estimator/classifier/SVC/templates/js/separated.method.txt#L98) is using addition in the same place. I'm guessing both are the same if the intercepts are having opposite sign, but, I'm not sure. Could you please shed some light on this?

**Comment by nok** [comment]
Hello @anmolshkl , do you get odd predictions between the original and transpiled estimator? Can you post the data and your params? Nevertheless there isn't any error in the regression tests of that classifier (e.g. [log](https://travis-ci.org/nok/sklearn-porter/jobs/311004004#L993-L1044)). But that doesn't mean, that there isn't any error.

**Comment by anmolshkl** [comment]
@nok thanks for the response. I'm getting the right predictions, and, the accuracy of the transpiled estimator is same as the scikit estimator (frankly, that's why I'm a little confused). To give you a little context about my confusion - I was trying to implement my own estimator for SVC and I tried following libsvm's code for prediction but I just couldn't get it right. Then, I looked into your estimator and found out that sklearn-porter's transpiled estimator uses [`decisions[d] = tmp + inters[d]`](https://github.com/nok/sklearn-porter/blob/master/sklearn_porter/estimator/classifier/SVC/templates/js/separated.method.txt#L98) while libsvm uses [`sum -= model->rho[p]`](https://github.com/cjlin1/libsvm/blob/master/svm.cpp#L2555). I replaced subtraction with an addition, and, my code started predicting values as expected.

I was just curious and wanted to know if you also encountered this problem while writing the algorithm for the estimator?

**Comment by nok** [comment]
Today I don't know how I figured it out in the past, but in general I add console outputs in the original scikit-learn classes and the generated classes. Then I compared each output and fix any differences on the fly. Finally I wrote regression tests to cover changes in the future.
